stdlib/3/tempfile.pyi

```diff
@@ -20,21 +21,19 @@ def TemporaryFile(
 def NamedTemporaryFile(
             mode: str = ..., buffering: int = ..., encoding: str = ...,
             newline: str = ..., suffix: str = ..., prefix: str = ...,
-            dir: str = ..., delete=...) -> BinaryIO:
+            dir: str = ..., delete: bool =...) -> BinaryIO:
     ...
 def SpooledTemporaryFile(
             max_size: int = ..., mode: str = ..., buffering: int = ...,
             encoding: str = ..., newline: str = ..., suffix: str = ...,
             prefix: str = ..., dir: str = ...) -> BinaryIO:
     ...
 
-class TemporaryDirectory:
+class TemporaryDirectory(ContextManager):
```

Comment:
I prefer not to do this -- contextlib.ContextManager is a lie (it doesn't exist in the stdlib, only in the stubs, and we're thinking of moving it to typing), and you don't need to inherit from it in order to be acceptable in a with-statement.